### ISSUE TITLE
Customer issue: "Avoid return types on setters"

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/linter#297]
This is currently occurring internally and causing false positives.

```dart
String _foo;

set foo(String foo) {
  if (_foo == foo) return; // Lint unhappy but shouldn't be.

  _foo = foo;
  // do stuff
}
```

**Comment by alexeieleusis** [comment]
What is the error you're getting? I can't reproduce in the unit test, copied your example in test/rules/avoid_return_types_on_setters.dart and ran with dart test/util/solo_test.dart avoid_return_types_on_setters, test passes.

**Comment by alexeieleusis** [comment]
FWIW I ran it with all rules enabled and the one that triggered on te setter is "Document all public members".

**Comment by matanlurey** [comment]
It's possible that this is internal only, maybe due to how we are invoking the linter.

I'll link you offline.

**Comment by alexeieleusis** [comment]
I would suggest closing this as not reproducible for now and reopen if needed.

**Comment by matanlurey** [comment]
Why? It's clearly reproducible, just perhaps not in this particular test case.
